app/helpers/sessions_helper.rb

```diff
@@ -15,4 +15,14 @@ def log_out
     session.delete :user_id
     @current_user = nil
   end
+
+  def remove_book_from_cart
+    session[:book_ids].delete(params[:book_id])
```

Comment:
Mạnh mẽ là tốt nhưng hơi lố rồi 
```suggestion
    session[:book_ids].delete(params[:book_id]) if session[:book_ids].include? params[:book_id]
```